### ISSUE TITLE
Feature: Cover extensions in standalone mode scenario - Have the ability to run an extension as a standalone ui mode

### ISSUE THREAD

**Issue by aalves08** [rancher/dashboard#11283]
Cover extensions in a standalone mode scenario. Have the ability to run an extension as a standalone ui mode.

As the first initiative, the changes might be profound on Rancher Dashboard to cover this, so it would be the bulk of the work.

- When you create a new extension, you get a skeleton app with it which is currently used for development
- We want to be able to build that app in the same way we build dashboard to give you a standalone application
- This, however, assumes that we are in a 'Rancher' world, i.e. you have the Rancher backend
- One of the goals here is to work without the Rancher backend:
   - Different or no authentication
   - Different on no settings, preferences
   - ...
- Will also want to customise the appearance - e.g. no side bar, different logo (maybe done via other mechanism anyway) etc

We need to think about being able to configure the app and swap out some layers - e.g. authentication, settings etc. We should look at what API calls the app makes as part of its 'bootstrap' and understand how to make those pluggable, so we can get the required info from different endpoints or not get it at all.

There may be a bit we can do here with Webpack.

You can imagine three use cases to drive this:

1. Empty =  'clean/empty shell' - no auth etc
2. Rancher = as is with Dashboard
3. NeuVector = using NV authentication/backend

**Comment by richard-cox** [comment]
Note this is already possible and used by Harvester and Epinio. There some tweaks in both, however both could log in and land directly in a `cluster` view

**Comment by aalves08** [comment]
I agree in terms of the use cases @richard-cox . If this in fact does proceed, my idea would be for this to become kind of a boilerplate solution where have this stripped down version of Dashboard neatly adjusted to cover "any" extension that we might throw at it, which is a bigger task overall but definitely future-proof.

**Comment by richard-cox** [comment]
dfsdf

**Comment by richard-cox** [comment]
Questions required for use cases

1. What is the core requirements of the functionality that needs to be supported? What problems does it need to solve?
1. What parts of the Rancher UI are users not expected to see or use?
1. In a standalone view, how important is authentication (logging in) and authorisation (global roles, and cluster and project members)?
1. How important is it that users cannot access the rancher instance (visit the ui, make api requests to it, etc)?
1. In a standalone view, how important is the application being hosted outside of a Rancher instance (two separate applications)?

Use cases
- Longhorn
- Security extensions
